internal/tools/pipeline/spec/pipeline_task.go

```diff
@@ -393,10 +394,22 @@ func (pt *PipelineTask) Convert2PB() *basepb.PipelineTaskDTO {
 	task.Result.Events = pt.Inspect.Events
 	task.Result.Errors = pt.MergeErrors2PB()
 	// handle metadata
+	newMetadata := make([]*commonpb.MetadataField, 0, len(task.Result.Metadata))
 	for _, field := range task.Result.Metadata {
 		field.Level = field.GetLevel()
+		// 将等级为ERROR的metadata添加到Errors中
+		if field.Level == string(metadata.MetadataLevelError) {
```

Comment:
Do we need to directly add the logic to expose pipeline errors? Please confirm. @chengjoey

Reply:
these are pipeline task errors, pipeline errors

@CeerDecy , why add metadata errors to `Results.Errors`, I think metadata and errors will both be displayed in UI

Reply:
just need to show up in `metadata`, we should let UI show red color if level is `error`

this pr just need add pre-checks information in pipeline task

Reply:
@iutx done